src/sentry/models/grouphashmetadata.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, JSONField, Model, region_silo_model
+
+
+@region_silo_model
+class GroupHashMetadata(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    class GroupingMethod(models.TextChoices):
+        HASH = "hash"
+        SEER = "seer"
+
+    class HashBasis(models.TextChoices):
+        MESSAGE = "msg"
+        FULL_STACKTRACE = "stack"
+        APP_STACKTRACE = "app_stack"
+        FINGERPRINT = "fingerprint"
+        # TODO: Are there others?
+
+    # GENERAL
+    grouphash = FlexibleForeignKey("sentry.GroupHash", unique=True, related_name="metadata")
+    date_added = models.DateTimeField(default=timezone.now)
+    grouping_method = models.PositiveIntegerField(choices=GroupingMethod)
+
+    # HASHING
+    # Most recent config to produce this hash
+    # TODO: do we need first, also?
+    # TODO: Is there a way to get choices for this from grouping.strategies.configurations.CONFIGURATIONS?
+    grouping_config = models.CharField()
```

Comment:
Beware: I'm handwaving here!

Maybe something like this:
```
from foo import CONFIGURATIONS

class ProjectConfig(models.TextChoices):
        CONFIG_0 = CONFIGURATIONS[0]
        CONFIG_1 = CONFIGURATIONS[1]
        ...
        CONFIG_N = CONFIGURATIONS[N]
```

Also, we should add an assertion for the size of it somewhere.

Reply:
That makes sense. I think the big issue here is going to be circular dependencies - lots of the grouping modules import from `models`, and in this model I'd have to be importing from one of those modules. Even if it technically works right now (because whatever I need doesn't happen to come from modules which import from `models`), I'd be nervous to rely on that continuing to the the case. So move it to a neutral location? But then it's populated by grouping code...